packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -7,6 +9,44 @@ import { LinuxParameters } from './linux-parameters';
 import { LogDriver, LogDriverConfig } from './log-drivers/log-driver';
 
 /**
+ * Properties for a Secret
+ */
+export interface SecretProps {
+  /**
+   * A SSM parameter that will be retrieved at container startup.
+   */
+  readonly parameter?: ssm.IParameter;
+
+  /**
+   * An AWS Secrets Manager secret that will be retrieved at container startup.
+   */
+  readonly secret?: secretsmanager.ISecret
+}
+
+/**
+ * A secret environment variable.
+ */
+export class Secret {
+  /**
+   * Creates an environment variable value from a parameter stored in AWS
+   * Systems Manager Parameter Store.
+   */
+  public static fromSsmParameter(parameter: ssm.IParameter) {
+    return new Secret({ parameter });
+  }
+
+  /**
+   * Creates a environment variable value from a secret stored in AWS Secrets
+   * Manager.
+   */
+  public static fromSecretsManager(secret: secretsmanager.ISecret) {
+    return new Secret({ secret });
+  }
+
+  constructor(public readonly props: SecretProps) {}
```

Comment:
This is a bit unnecessarily convoluted I think. Why not just accept an “arn” string here? This will remove the need for SecretProps and the special casing in renderContainerDefintion

Reply:
How do you suggest granting read (`secret.grantRead()` or `parameter.grantRead()`) to the task execution role in `renderContainerDefinition`? With a `.fromArn()` there (there's no way to import from ARN currently in SSM, works for AWS Secrets)

Reply:
It also possible to do `taskDefintion.obtainExecutionRole().addToPolicy(...)` but I find that duplicating SSM/Secrets read IAM permissions is not a elegant solution...

Reply:
Something like that (polymorphism ;-)):

```ts
abstract class Secret {

  public static fromSecretsManager(secret: ISecret): Secret {
    return {
      arn: secret.secretArn,
      grantRead: grantee => secret.grantRead(grantee)
    };
  }
  
  public abstract readonly arn: string;
  public abstract grantRead(grantee: IGrantee): Grant;
}
```

Reply:
Nice